### ISSUE TITLE
Improve dialog box message for failed pull when affected locally modified files exist

### ISSUE THREAD

**Issue by chrispy-snps** [oxygenxml/oxygen-git-client-addon#172]
When all of the following are true:

* There are locally modified files (staged or unstaged)
* There are unpulled upstream commits
* There are no unpushed local commits

then a `git pull --ff-only` returns the following:

```
$ git pull --ff-only
...
error: Your local changes to the following files would be overwritten by merge:
        dita/test.dita
Please commit your changes or stash them before you merge.
Aborting
```

The equivalent Git add-on dialog box in Oxygen is as follows:

![image](https://user-images.githubusercontent.com/50950969/214436275-93e6ad27-d487-4c8a-a370-67aeee6b4c9c.png)

In Git, this word typically means that there are irreconcilable changes on the same or adjacent lines. This sometimes confuses our writers because they will change a single topic reference in a large .ditamap file, then get this message, then wonder why someone else is also working on their topic.

However in this case, Git cannot pull a remote file onto a locally modified file regardless of what the changes are - whether they conflict, don't conflict, or even if they are identical!

The command-line Git message does not use the word "conflict". Perhaps the add-on's dialog text could use similar wording:

> Cannot pull (%s) changes because your local uncommitted changes to the files below would be overwritten by the incoming remote changes. (These changes might not conflict, but changes in uncommitted files cannot be merged.)
> Commit, stash, or discard the local changes and try again.

**Comment by AlexJitianu** [comment]
Hi Chris,
I understand and thank you for the suggestion. We will improve the message in the next release.

**Comment by sorincarbunaru** [comment]
Note: internal issue ID is EXM-52307.

**Comment by AlexSmarandache15** [comment]
Hi @chrispy-snps,

Do you think this message is suitable for both types of pull (rebase and merge)? Or would it only be suitable for rebase?

> Cannot pull (%s) changes because your local uncommitted changes to the files below would be overwritten by the incoming remote changes. (These changes might not conflict, but changes in uncommitted files cannot be merged.)
> Commit, stash, or discard the local changes and try again.

**Comment by chrispy-snps** [comment]
@AlexSmarandache15 - I repeated the experiment with a pull-mere. I get a similar message from command-line Git:

```
$ git pull
...
error: Your local changes to the following files would be overwritten by merge:
        dita/test1.dita
Please commit your changes or stash them before you merge.
Aborting
```

and from the Oxygen Git add-on:

![image](https://user-images.githubusercontent.com/50950969/224360794-00e89e19-b6d0-4eb2-8e8a-ed8da8ed7a17.png)

and I also confirmed that the message occurs whether the files conflict, don't conflict, or even if they are identical.

So yes, this proposed message would equally apply to both pull-rebase and pull-merge.

Thank you for asking this question!
